src/main/scala/io/iohk/ethereum/consensus/ethash/EthashMiner.scala

```diff
@@ -93,7 +93,7 @@ class EthashMiner(
         val startTime = System.currentTimeMillis()
         val mineResult = mine(headerHash, block.header.difficulty.toLong, dagSize, dag, miningConfig.mineRounds)
         val time = System.currentTimeMillis() - startTime
-        val hashRate = (mineResult.triedHashes * 1000) / time
+        val hashRate = if (time > 0) (mineResult.triedHashes * 1000) / time else Long.MaxValue
```

Comment:
I was wondering if we could use a "better" (=monotonic) clock, e.g. `System.nanoTime()` whose purpose is to measure elapsed time. There is no strong guarantee that this is indeed going to be monotonic because it ultimately depends on the OS facilities but this is generally not a problem in Linux nowadays. 

Another possible route is to not update the hash rate if we cannot reliably compute it.

Reply:
Note: guava has a `com.google.common.base.StopWatch` class for this purpose.

Reply:
I'd rather not spend too much time on this, this is not really a big problem anyway. I can add a FIXME though